### PR TITLE
Bb chapter 2

### PR DESCRIPTION
**Pull request by mb706**

### CODE REVIEW

File: book/chapters/chapter2/data_and_basic_modeling.qmd [hunk 1]
```diff
@@ -8,33 +8,35 @@
...
+The term 'task' is used to refer to the prediction problem that we are trying to solve.
 Tasks are defined by the features used for prediction and the targets to predict, so there can be multiple tasks associated with any given dataset.
 For example, predicting miles per gallon (mpg) from horsepower is one task, predicting horsepower from mpg is another task, and predicting the number of gears from the car's model is yet another task.
 
 Supervised learning can be further divided into `r index('regression', aside = TRUE)` -- which is the prediction of numeric target values, e.g. predicting a car's mpg -- and `r index('classification', aside = TRUE)` -- which is the prediction of categorical values/labels, e.g., predicting a car's model.
-Other tasks are also encompassed by supervised learning, and we return to them in @sec-special, which also considers `r index('unsupervised learning')` tasks.
+@sec-special also discusses other tasks, such `r index('cost-sensitive classification', "cost-sensitive", parent = "classification")` or `r index('unsupervised learning')`.
```
Comment (RaphaelS1) [inline book/chapters/chapter2/data_and_basic_modeling.qmd:20]:
```suggestion
@sec-special also discusses other tasks, including `r index('cost-sensitive classification', "cost-sensitive", parent = "classification")` and `r index('unsupervised learning')`.
```

File: book/chapters/chapter2/data_and_basic_modeling.qmd [hunk 2]
```diff
@@ -8,33 +8,35 @@
...
 For any supervised learning task, the goal is to build a `r index('model')` that captures the relationship between the features and target, often with the goal of `r index('training', "model training")` the model to learn relationships about the data so it can make predictions for new and previously unseen data.
-A `r index('model', aside = TRUE)` is formally a mapping from a feature vector to predictions, such models are induced by passing `r index('training data')` to machine learning algorithms and include `r index('decision trees', "decision tree")`, `r index('support vector machines', "support vector machine")`, `r index('neural networks', "neural network")`, and many more.
+A `r index('model', aside = TRUE)` is formally a mapping from a feature vector to a prediction.
+A prediction can take many forms depending on the task; for classification, it can be a predicted label, or a set of predicted probabilities or scores.
```
Comment (RaphaelS1) [inline book/chapters/chapter2/data_and_basic_modeling.qmd:23]:
```suggestion
A prediction can take many forms depending on the task; for example, in classification this can be a predicted label, or a set of predicted probabilities or scores.
```

File: book/chapters/chapter2/data_and_basic_modeling.qmd [hunk 3]
```diff
@@ -98,7 +100,7 @@ The `id` argument is optional and specifies an identifier for the task that is u
 
 As many machine learning models do not work properly with arbitrary UTF8 names, `mlr3` defaults to throwing an error if any of the column names passed to `r ref("as_task_regr()")` (and other task constructors) contain a non-ASCII character or do not comply with R's variable naming scheme.
 Therefore, we recommend converting names with `r ref("make.names()")` if possible.
-You can bypass this check by setting `options(mlr3.allow_utf8_names = TRUE)` (but do not be surprised if errors occur).
+You can bypass this check by setting `options(mlr3.allow_utf8_names = TRUE)` (but do not be surprised if errors occur later, especially when passing objects down to other packages).
```
Comment (RaphaelS1) [inline book/chapters/chapter2/data_and_basic_modeling.qmd:103]:
```suggestion
You can bypass this check by setting `options(mlr3.allow_utf8_names = TRUE)` (but do not be surprised if errors occur later, especially when passing objects to other packages).
```

File: book/chapters/chapter2/data_and_basic_modeling.qmd [hunk 4]
```diff
@@ -379,9 +380,8 @@ Having covered the unified train/predict interface, we can now look at how to us
 
 `Learner`s encapsulate a machine learning algorithm and its `r index('hyperparameters')`, which affect *how* the algorithm is run and can be set by the user.
 Hyperparameters may affect how a model is trained or how it makes predictions and deciding how to set hyperparameters can require expert knowledge.
-Hyperparameters are usually not manually configured but instead are automatically optimized (@sec-optimization).
-However, for this chapter, we will only discuss manual setting of hyperparameters for the sake of brevity.
-We will first look at `r ref_pkg("paradox")` and `r ref("ParamSet")` objects which are used to store learner hyperparameters, and then we will look at getting and setting these values.
+Hyperparameters can be optimized automatically (@sec-optimization), however, in this chapter, we show how to set them manually.
+However, for this chapter, we will only discuss manual setting of hyperparameters.
```
Comment (RaphaelS1) [inline book/chapters/chapter2/data_and_basic_modeling.qmd:384]:
```suggestion
```

File: book/chapters/chapter2/data_and_basic_modeling.qmd [hunk 5]
```diff
@@ -379,9 +380,8 @@ Having covered the unified train/predict interface, we can now look at how to us
 
 `Learner`s encapsulate a machine learning algorithm and its `r index('hyperparameters')`, which affect *how* the algorithm is run and can be set by the user.
 Hyperparameters may affect how a model is trained or how it makes predictions and deciding how to set hyperparameters can require expert knowledge.
-Hyperparameters are usually not manually configured but instead are automatically optimized (@sec-optimization).
-However, for this chapter, we will only discuss manual setting of hyperparameters for the sake of brevity.
-We will first look at `r ref_pkg("paradox")` and `r ref("ParamSet")` objects which are used to store learner hyperparameters, and then we will look at getting and setting these values.
+Hyperparameters can be optimized automatically (@sec-optimization), however, in this chapter, we show how to set them manually.
```
Comment (RaphaelS1) [inline book/chapters/chapter2/data_and_basic_modeling.qmd:383]:
```suggestion
Hyperparameters can be optimized automatically (@sec-optimization), but in this chapter we will focus on how to set them manually.
```

File: book/chapters/chapter2/data_and_basic_modeling.qmd [hunk 6]
```diff
@@ -392,18 +392,18 @@ To access the hyperparameters in the decision tree, we use `r index("$param_set"
...
-| `r ref("ParamDbl", index = TRUE)`  | Real-valued (Numeric) Parameters     |
-| `r ref("ParamInt", index = TRUE)`  | Integer Parameters                   |
-| `r ref("ParamFct", index = TRUE)`  | Categorical (Factor) Parameters      |
-| `r ref("ParamLgl", index = TRUE)`  | Logical / Boolean Parameters         |
-| `r ref("ParamUty", index = TRUE)`  | Untyped Parameters                   |
+| `r ref("ParamDbl", index = TRUE)`  | Real-valued (numeric) parameters     |
+| `r ref("ParamInt", index = TRUE)`  | Integer parameters                   |
+| `r ref("ParamFct", index = TRUE)`  | Categorical (factor) parameters      |
+| `r ref("ParamLgl", index = TRUE)`  | Logical / Boolean parameters         |
+| `r ref("ParamUty", index = TRUE)`  | Untyped parameters                   |
```
Comment (RaphaelS1) [inline book/chapters/chapter2/data_and_basic_modeling.qmd:406]: Maybe change column 2 title to "Parameter Type" and then you don't need to write "parameters" each time in the columns?

File: book/chapters/chapter2/data_and_basic_modeling.qmd [hunk 7]
```diff
@@ -413,14 +413,14 @@ In our decision tree example, we can infer from the `ParamSet` output that:
 * `keep_model` must be a "logical" (`ParamLgl`) taking values `TRUE` or `FALSE` with default `FALSE`
 * `xval` must be an "integer" (`ParamInt`) taking values between `0` and `Inf` with a default of `10` and has a set value of `0`.
 
-In rare cases (we try to minimize it as much as possible), we alter hyperparameter values when constructing a learner.
-When we do this, the reason will always be given in the learner help page.
-In the case of `lrn("regr.rpart")`, we changed the `xval` default to `0` because `xval` controls internal cross-validations and if a user accidentally leaves this at the default `10`, model training can take a long time.
+In rare cases (`mlr3` tries to minimize it as much as possible), hyperparameters are initialized to values which deviate from the default in the underlying package.
```
Comment (RaphaelS1) [inline book/chapters/chapter2/data_and_basic_modeling.qmd:416]: I think an active voice makes more sense. It isn't `mlr3` that decides on the changes it is us making an active decision. Currently it sounds like software makes this decision

File: book/chapters/chapter2/data_and_basic_modeling.qmd [hunk 8]
```diff
@@ -736,7 +734,7 @@ lrn_rpart$predict(tsk_penguins, splits$test)$score(measure)
 ```
 
 In this experiment, we loaded the predefined task `penguins`, which is based on the `r ref("palmerpenguins::penguins")` dataset, then partitioned the data into training and test splits.
-We loaded the featureless classification baseline (using the default which always predicts the most common class in the training data) and a classification decision tree, then the accuracy measure (number of correct predictions divided by total number of predictions), trained our models and finally made and scored predictions.
+We loaded the featureless classification baseline (using the default which always predicts the most common class in the training data, but which also has the option of predicting random response values) and a classification decision tree, then the accuracy measure (number of correct predictions divided by total number of predictions), trained our models and finally made and scored predictions.
```
Comment (RaphaelS1) [inline book/chapters/chapter2/data_and_basic_modeling.qmd:737]: Random draws but from the empirical distribution right?

File: book/chapters/chapter2/data_and_basic_modeling.qmd [hunk 9]
```diff
@@ -825,26 +819,27 @@ prediction
...
+We need to use `sapply` for this, the `task_properties` column is a list column.
 We can evaluate the quality of our probability predictions and response predictions simultaneously by providing multiple measures:
 
 ```{r}
 measures = msrs(c("classif.mbrier", "classif.logloss", "classif.acc"))
 prediction$score(measures)
 ```
 
-The `r index('accuracy')` measure evaluates the `"response"` predictions whereas the `r index('brier score')` (`"classif.mbrier"`) (squared difference between predicted probabilities and the truth) and `r index('logloss')` (`"classif.logloss"`) (negative logarithm of the predicted probability for the true class) are evaluating the probability predictions.
+The `r index('accuracy')` measure evaluates the `"response"` predictions whereas the `r index('Brier score')` (`"classif.mbrier"`, squared difference between predicted probabilities and the truth) and `r index('logloss')` (`"classif.logloss"`, negative logarithm of the predicted probability for the true class) are evaluating the probability predictions.
```
Comment (RaphaelS1) [inline book/chapters/chapter2/data_and_basic_modeling.qmd:842]:
```suggestion
The `r index('accuracy')` measure evaluates the `"response"` predictions whereas the `r index('Brier score', lower = FALSE)` (`"classif.mbrier"`, squared difference between predicted probabilities and the truth) and `r index('logloss')` (`"classif.logloss"`, negative logarithm of the predicted probability for the true class) are evaluating the probability predictions.
```

File: book/chapters/chapter2/data_and_basic_modeling.qmd [hunk 10]
```diff
@@ -900,9 +895,9 @@ We will return to the concept of binary (mis)classification in greater detail in
 
 #### Thresholding {.unnumbered .unlisted}
 
-The final big difference to discuss is `r index('thresholding', aside = TRUE)`.
+The final big difference with regression to discuss is `r index('thresholding', aside = TRUE)`.
```
Comment (RaphaelS1) [inline book/chapters/chapter2/data_and_basic_modeling.qmd:898]:
```suggestion
The final big difference compared to regression we will discuss is `r index('thresholding', aside = TRUE)`.
```

File: book/chapters/chapter2/data_and_basic_modeling.qmd [hunk 11]
```diff
@@ -937,7 +932,7 @@ plt = plt + ggplot2::scale_fill_manual(values = c("grey30", "grey50"))
 print(plt)
 ```
 
-Whilst this model may appear to have good performance on the surface, in fact, it just ignores all 'bad' customers -- this can create very big problems in healthcare and other settings where there are data biases, or for insurance companies if `r index('false positives')` cost more than `r index('false negatives')` (see @sec-cost-sens for cost-sensitive classification).
+Whilst this model may appear to have good performance on the surface, in fact, it just ignores all 'bad' customers -- this can also create very big problems here, or in healthcare tasks and other settings where `r index('false positives')` cost more than `r index('false negatives')` (see @sec-cost-sens for cost-sensitive classification).
```
Comment (RaphaelS1) [inline book/chapters/chapter2/data_and_basic_modeling.qmd:935]:
```suggestion
Whilst this model may appear to have good performance on the surface, in fact, it just ignores all 'bad' customers -- this can create big problems in this finance example, as well as in healthcare tasks and other settings where `r index('false positives')` cost more than `r index('false negatives')` (see @sec-cost-sens for cost-sensitive classification).
```

File: book/chapters/chapter2/data_and_basic_modeling.qmd [hunk 12]
```diff
@@ -392,18 +391,18 @@ To access the hyperparameters in the decision tree, we use `r index("$param_set"
...
 ```
 
-The output above is a `r ref("ParamSet", aside = TRUE)` object.
-These objects provide information on hyperparameters including their name (`id`), data type (`class`), acceptable ranges for hyperparameter values (`lower`, `upper`), the number of levels possible if the data type is categorical (`nlevels`), the default value from the underlying package (`default`), and finally the set value (`value`).
+The output above is a `r ref("ParamSet", aside = TRUE)` object, supplied by the `r ref_pkg("paradox")` package.
+These objects provide information on hyperparameters including their name (`id`), data type (`class`), technically valid ranges for hyperparameter values (`lower`, `upper`), the number of levels possible if the data type is categorical (`nlevels`), the default value from the underlying package (`default`), and finally the set value (`value`).
 The second column references classes defined in `r ref_pkg("paradox")` that determine the class of the parameter and the possible values it can take.
 @tbl-parameters-classes lists the possible hyperparameter types, all of which inherit from `r ref("Param")`.
 
 | Hyperparameter Class | Description                          |
```
Comment (RaphaelS1) [inline book/chapters/chapter2/data_and_basic_modeling.qmd:399]:
```suggestion
| Hyperparameter Class | Hyperparameter Type                          |
```

File: book/chapters/chapter2/data_and_basic_modeling.qmd [hunk 13]
```diff
@@ -392,18 +391,18 @@ To access the hyperparameters in the decision tree, we use `r index("$param_set"
...
-| `r ref("ParamDbl", index = TRUE)`  | Real-valued (Numeric) Parameters     |
-| `r ref("ParamInt", index = TRUE)`  | Integer Parameters                   |
-| `r ref("ParamFct", index = TRUE)`  | Categorical (Factor) Parameters      |
-| `r ref("ParamLgl", index = TRUE)`  | Logical / Boolean Parameters         |
-| `r ref("ParamUty", index = TRUE)`  | Untyped Parameters                   |
+| `r ref("ParamDbl", index = TRUE)`  | Real-valued (numeric) parameters     |
+| `r ref("ParamInt", index = TRUE)`  | Integer parameters                   |
+| `r ref("ParamFct", index = TRUE)`  | Categorical (factor) parameters      |
+| `r ref("ParamLgl", index = TRUE)`  | Logical / Boolean parameters         |
+| `r ref("ParamUty", index = TRUE)`  | Untyped parameters                   |
```
Comment (RaphaelS1) [inline book/chapters/chapter2/data_and_basic_modeling.qmd:405]:
```suggestion
| `r ref("ParamDbl", index = TRUE)`  | Real-valued (numeric)     |
| `r ref("ParamInt", index = TRUE)`  | Integer                   |
| `r ref("ParamFct", index = TRUE)`  | Categorical (factor)      |
| `r ref("ParamLgl", index = TRUE)`  | Logical / Boolean         |
| `r ref("ParamUty", index = TRUE)`  | Untyped                   |
```

File: book/chapters/chapter2/data_and_basic_modeling.qmd [hunk 14]
```diff
@@ -413,14 +412,14 @@ In our decision tree example, we can infer from the `ParamSet` output that:
 * `keep_model` must be a "logical" (`ParamLgl`) taking values `TRUE` or `FALSE` with default `FALSE`
 * `xval` must be an "integer" (`ParamInt`) taking values between `0` and `Inf` with a default of `10` and has a set value of `0`.
 
-In rare cases (we try to minimize it as much as possible), we alter hyperparameter values when constructing a learner.
-When we do this, the reason will always be given in the learner help page.
-In the case of `lrn("regr.rpart")`, we changed the `xval` default to `0` because `xval` controls internal cross-validations and if a user accidentally leaves this at the default `10`, model training can take a long time.
+In rare cases (`mlr3` tries to minimize it as much as possible), hyperparameters are initialized to values which deviate from the default in the underlying package.
```
Comment (RaphaelS1) [inline book/chapters/chapter2/data_and_basic_modeling.qmd:415]:
```suggestion
In rare cases (we try to minimize it as much as possible), hyperparameters are initialized to values which deviate from the default in the underlying package.
```

File: book/chapters/chapter2/data_and_basic_modeling.qmd [hunk 15]
```diff
@@ -736,7 +733,7 @@ lrn_rpart$predict(tsk_penguins, splits$test)$score(measure)
 ```
 
 In this experiment, we loaded the predefined task `penguins`, which is based on the `r ref("palmerpenguins::penguins")` dataset, then partitioned the data into training and test splits.
-We loaded the featureless classification baseline (using the default which always predicts the most common class in the training data) and a classification decision tree, then the accuracy measure (number of correct predictions divided by total number of predictions), trained our models and finally made and scored predictions.
+We loaded the featureless classification baseline (using the default which always predicts the most common class in the training data, but which also has the option of predicting random response values) and a classification decision tree, then the accuracy measure (number of correct predictions divided by total number of predictions), trained our models and finally made and scored predictions.
```
Comment (RaphaelS1) [inline book/chapters/chapter2/data_and_basic_modeling.qmd:736]:
```suggestion
We loaded the featureless classification baseline (using the default which always predicts the most common class in the training data, but which also has the option of predicting (unifomly or weighted) random response values) and a classification decision tree, then the accuracy measure (number of correct predictions divided by total number of predictions), trained our models and finally made and scored predictions.
```

File: book/chapters/chapter2/data_and_basic_modeling.qmd [hunk 16]
```diff
@@ -736,7 +733,7 @@ lrn_rpart$predict(tsk_penguins, splits$test)$score(measure)
 ```
 
 In this experiment, we loaded the predefined task `penguins`, which is based on the `r ref("palmerpenguins::penguins")` dataset, then partitioned the data into training and test splits.
-We loaded the featureless classification baseline (using the default which always predicts the most common class in the training data) and a classification decision tree, then the accuracy measure (number of correct predictions divided by total number of predictions), trained our models and finally made and scored predictions.
+We loaded the featureless classification baseline (using the default which always predicts the most common class in the training data, but which also has the option of predicting (unifomly or weighted) random response values) and a classification decision tree, then the accuracy measure (number of correct predictions divided by total number of predictions), trained our models and finally made and scored predictions.
```
Comment (RaphaelS1) [inline book/chapters/chapter2/data_and_basic_modeling.qmd:736]:
```suggestion
We loaded the featureless classification baseline (using the default which always predicts the most common class in the training data, but which also has the option of predicting (uniformly or weighted) random response values) and a classification decision tree, then the accuracy measure (number of correct predictions divided by total number of predictions), trained our models and finally made and scored predictions.
```
